mico-core/src/main/java/io/github/ust/mico/core/service/GitHubCrawler.java

```diff
@@ -53,48 +57,96 @@ public GitHubCrawler(RestTemplate restTemplate, KubernetesNameNormalizer kuberne
+        ResponseEntity<String> responseBasicInfo;
+        ResponseEntity<String> responseReleaseInfo;
+
+        try {
+            responseBasicInfo = restTemplate.getForEntity(uriBasicInfo, String.class);
+        } catch (Exception e) {
+            throw new IllegalArgumentException("GitHub repository " + uriBasicInfo.replace(GITHUB_API_URL, "") + " does not exist!");
+        }
 
         try {
+            responseReleaseInfo = restTemplate.getForEntity(uriReleaseInfo, String.class);
+        } catch (Exception e) {
+            throw new IllegalArgumentException("GitHub repository " + uriBasicInfo.replace(GITHUB_API_URL, "")
+                + " doesn't have a release " + uriReleaseInfo.replace(GITHUB_API_URL, "") + "!");
+        }
+
+        if (responseBasicInfo.getStatusCode().is2xxSuccessful() && responseReleaseInfo.getStatusCode().is2xxSuccessful()) {
+            ObjectMapper mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
             JsonNode basicInfoJson = mapper.readTree(responseBasicInfo.getBody());
             JsonNode releaseInfoJson = mapper.readTree(responseReleaseInfo.getBody());
 
             String name = basicInfoJson.get("name").textValue();
             String normalizedName = kubernetesNameNormalizer.normalizeName(name);
 
-            return new MicoService()
+            MicoService micoService = new MicoService()
                 .setShortName(normalizedName)
                 .setName(basicInfoJson.get("full_name").textValue())
                 .setVersion(releaseInfoJson.get("tag_name").textValue())
                 .setDescription(basicInfoJson.get("description").textValue())
                 .setServiceCrawlingOrigin(MicoServiceCrawlingOrigin.GITHUB)
                 .setGitCloneUrl(basicInfoJson.get("clone_url").textValue())
                 .setGitReleaseInfoUrl(releaseInfoJson.get("url").textValue());
+            if (!StringUtils.isEmpty(dockerfilePath)) {
+                if (existsFileInGithubRepo(uriBasicInfo, dockerfilePath)) {
+                    micoService.setDockerfilePath(dockerfilePath);
+                } else {
+                    throw new IllegalArgumentException("The dockerfile path must be a valid path relativ the the repository root");
+                }
+            }
+            return micoService;
+        } else {
+            throw new IllegalArgumentException("An error occurred while requesting information about the github repository. Send to requests and got the status codes "
+                + responseBasicInfo.getStatusCode() + " and " + responseReleaseInfo.getStatusCode().is2xxSuccessful());
+        }
 
-        } catch (IOException e) {
-            log.error(e.getStackTrace().toString());
-            log.error("Getting exception '{}'", e.getMessage());
-            throw e;
+    }
+
+    /**
+     * Checks if a file in a github repository exists
+     * @param uriBasicInfo
+     * @param dockerfilePath
+     * @return
```

Comment:
Add JavaDoc for parameters and return statement.

Reply:
Done